android/build.gradle

```diff
@@ -1033,13 +1033,13 @@ dependencies {
 
     if (REACT_NATIVE_MINOR_VERSION >= 71) {
         if (isPackageBuild()) {
-            implementation "com.facebook.react:react-android:+"
+            implementation "com.facebook.react:react-android:" + REACT_NATIVE_VERSION
         } else {
-            implementation "com.facebook.react:react-android:" // version substituted by RNGP
+            implementation "com.facebook.react:react-android:0.71.5" // version substituted by RNGP
```

Comment:
For RN 0.71+, the version should be substituted by react-native-gradle-plugin, at least when building an application. Also, leaving `implementation` without the version here should be safe as Gradle will throw an error if there's no version.